apps/entity-service/src/entities/dto/site-report.dto.ts

```diff
@@ -0,0 +1,302 @@
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated project uuid"
+  })
+  projectUuid: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation name"
+  })
+  organisationName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation uuid"
+  })
+  organisationUuid: string | null;
+
+  @ApiProperty()
+  updatedAt: Date;
+
+  @ApiProperty({ nullable: true })
+  submittedAt: Date | null;
+
+  @ApiProperty({ nullable: true })
+  taskId: number | null;
+
+  @ApiProperty()
+  dueAt: Date | null;
+
+  @ApiProperty({ nullable: true })
+  reportTitle: string | null;
+
+  @ApiProperty()
+  createdAt: Date;
+}
+
+export type AdditionalSiteReportLightProps = Pick<SiteReportLightDto, "reportTitle">;
+export type AdditionalSiteReportFullProps = AdditionalSiteReportLightProps &
+  AdditionalProps<SiteReportFullDto, SiteReportLightDto & Omit<SiteReport, "site">>;
+export type SiteReportMedia = Pick<SiteReportFullDto, keyof typeof SiteReport.MEDIA>;
+
+export class SiteReportFullDto extends SiteReportLightDto {
+  constructor(siteReport: SiteReport, props?: AdditionalSiteReportFullProps) {
+    super();
+    this.populate(SiteReportFullDto, {
+      ...pickApiProperties(siteReport, SiteReportFullDto),
+      lightResource: false,
+      // these two are untyped and marked optional in the base model.
+      createdAt: siteReport.createdAt as Date,
+      updatedAt: siteReport.createdAt as Date,
+      ...props
+    });
+  }
+
+  @ApiProperty({ nullable: true })
+  reportTitle: string | null;
+
+  @ApiProperty({ nullable: true })
+  projectReportTitle: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated site name"
+  })
+  siteName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated site uuid"
+  })
+  siteUuid: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation name"
+  })
+  organisationName: string | null;
+
+  @ApiProperty({
+    nullable: true,
+    description: "The associated organisation uuid"
+  })
+  organisationUuid: string | null;
+
+  @ApiProperty()
+  dueAt: Date | null;
+
+  @ApiProperty()
+  status: string;
+
+  @ApiProperty()
+  updateRequestStatus: string;
+
+  @ApiProperty({ nullable: true })
+  feedback: string | null;
+
+  @ApiProperty({ nullable: true })
+  feedbackFields: string[] | null;
+
+  @ApiProperty()
+  nothingToReport: boolean;
+
+  @ApiProperty()
+  readableCompletionStatus: string;
+
+  @ApiProperty({ nullable: true })
+  title: string | null;
+
+  @ApiProperty({ nullable: true })
+  sharedDriveLink: string | null;
+
+  @ApiProperty({ nullable: true })
+  createdBy: number | null;
+
+  @ApiProperty({ nullable: true })
+  createdByUser: User | null;
+
+  @ApiProperty({ nullable: true })
+  approvedBy: number | null;
+
+  @ApiProperty({ nullable: true })
+  approvedByUser: User | null;
```

Comment:
Including a bare User record in a DTO is an anti pattern. We should never be asking the JSON API serialization code to encode a full DB model. This should be a light weight DTO including just what's needed. Probably just the name fields? Also, are the `createdBy` and `approvedBy` numerical user IDs actually useful for the client? I wouldn't expect it to want them.

Reply:
Everything else looks resolved, but this still needs attention.